buildchain/buildchain/versions.py

```diff
@@ -228,132 +228,193 @@ def _version_prefix(version: str, prefix: str = 'v') -> str:
 CONTAINER_IMAGES_MAP = {image.name: image for image in CONTAINER_IMAGES}
 
 # }}}
+
 # Packages {{{
 
+class PackageVersion:
+    """A package's authoritative version data.
+
+    This class contains version information for a named package, and
+    provides helper methods for formatting version/release data as well
+    as fully versioned name, for all supported OS families.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        version: Optional[str] = None,
+        release: Optional[str] = None
+    ):
+        """Initializes a package version.
+
+        Arguments:
+            name: the name of the package
+            version: the version of the package
+            release: the release of the package
+        """
+        self._name = name
+        self._version = version
+        self._release = release
+
+    name = property(operator.attrgetter('_name'))
+    version = property(operator.attrgetter('_version'))
+    release = property(operator.attrgetter('_release'))
+
+    @property
+    def full_version(self) -> Optional[str]:
+        """The full package version string."""
+        full_version = None
+        if self.version:
+            full_version = self.version
+            if self.release:
+                full_version = '{}-{}'.format(self.version, self.release)
+        return full_version
+
+    @property
+    def rpm_full_name(self) -> str:
+        """The package's full name in RPM conventions."""
+        name_fmt = '{p.name}'
+        if self.version:
+            name_fmt += '-{p.version}'
+            if self.release:
+                name_fmt += '-{p.release}'
+        return name_fmt.format(p=self)
+
+    @property
+    def deb_full_name(self) -> str:
```

Comment:
ditto, could we do something like:

```
if self.version:
    return '{}={}'.format(self.name, self.full_version)
return self.name
```
?

Reply:
Oh indeed, that's basic DRY. Thanks.